R-packages/evalcast/R/manage_cards.R

```diff
@@ -91,3 +91,34 @@ intersect_averagers <- function(cards, grp_vars, avg_vars) {
   }
   cards %>% right_join(averagers_intersected)
 }
+
+#' Scale error measures based on those of a particular forecaster.
+#'
+#' @param score_card score_card like that returned by `evaluate_predictions()`
+#' @param vars vector of column names in `score_card` to normalize
+#' @param denom_forecaster name of forecaster in `score_card$forecaster` column by whose error
+#'   values the remaining forecasters' errors will be scaled
+#' @param err_cols vector of column names in `score_card` that contain error measures.
+#'   Elements of `err_cols` that are not in `vars` will be dropped from the final output to
+#'   avoid a mix of scaled and unscaled measures.
+#'
+#' @return
+#' @export
+scale_by_forecaster <- function(score_card, vars, denom_forecaster, 
+                                err_cols = c("ae", "wis")) {
+  df_list <- map(vars, function(var) {
+    score_card %>% 
+      select(setdiff(names(score_card), setdiff(err_cols, var))) %>% 
+      pivot_wider(names_from = "forecaster", 
+                  names_prefix = var, 
+                  values_from = var) %>% 
+      mutate(across(starts_with(var), ~ .x /
+                      !!sym(paste0(var, denom_forecaster)))) %>%
+      pivot_longer(cols = starts_with(var), 
+                   names_to = "forecaster",
+                   values_to = var) %>%
+      mutate(forecaster = substring(forecaster, nchar(var) + 1)) %>%
+      filter(forecaster != denom_forecaster)
+  })
+  return(reduce(df_list, left_join))
+}
```

Comment:
This will return NaN's (if the baseline score is 0) or NA's (if the baseline doesn't exist). I think that behavior is correct, but might warrant a warning.

Reply:
Good point. Yes, we should do something about this. To avoid the NaN, we could add a tolerance parameter so that we divide by `max(baseline, tol)`. I think the NA is probably the right thing if the baseline doesn't have a forecast, but should probably document this behavior, and throw a warning before the return.

Reply:
After writing more tests, the `across()` statement fails when all forecasts are from `baseline`.  It now throws an error if the baseline is not in the `forecaster` column or is the only entry.

Reply:
I think your changes address if there is no baseline forecaster, but not if the value is 0.

Reply:
Done

Reply:
The error text doesn't seem accurate, since division by 0 doesn't result in an error in R, it results in the value NaN. I think we should use a warning instead of an error, since a user may or may not want to keep the NaNs. Also, see dajmcdon's comment in this thread about using a tolerance parameter.

Reply:
Ok I have changed it to just be a warning but have let the NaN remain